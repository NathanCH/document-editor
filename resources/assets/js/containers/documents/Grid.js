import React from 'react';
import PropTypes from 'prop-types';
import GridItem from './GridItem';
import withLoader from './withLoader';

import './Grid.scss';

const Grid = props => (
  <div className="grid">
    {Object.keys(props.items).map(id => (
      <GridItem 
        item={props.items[id]} 
        key={props.items[id].id} />
    ))}
  </div>
);

Grid.propTypes = {
  items: PropTypes.object,
  isFetching: PropTypes.bool,
};

Grid.defaultProps = {
  items: {},
  isFetching: false,
};

export default withLoader(Grid);