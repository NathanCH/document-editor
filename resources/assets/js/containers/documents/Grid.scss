@import './resources/assets/sass/variables';

.grid {
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  grid-gap: 2rem;
    @media (max-width: $lg) {
      grid-template-columns: repeat(3, 1fr);
    }
    @media (max-width: $md) {
      grid-template-columns: repeat(2, 1fr);
    }
}