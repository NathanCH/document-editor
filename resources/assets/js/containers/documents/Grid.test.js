import React from 'react';
import renderer from 'react-test-renderer';
import { bindActionCreators } from 'redux';

import Grid from './Grid';

describe('<Grid />', () => {

  test('it renders without props', () => {
    const component = renderer
      .create(<Grid />)
      .toJSON();

    expect(component).toMatchSnapshot();
  });

  // test('it renders a grid item for each `props.items`', () => {

  // });

});