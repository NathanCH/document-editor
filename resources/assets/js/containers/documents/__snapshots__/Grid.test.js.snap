// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Grid /> it renders without props 1`] = `<div />`;
